### ISSUE TITLE
Don't invoke beforeSend for transactions

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#1730]
### Description

The SDK must not invoke `beforeSend` for transactions. If users want to drop transactions they should instead use event processors, see https://develop.sentry.dev/sdk/performance/#interaction-with-beforesend-and-event-processors

As some users might rely on calling `beforeSend` to drop transactions and event processors are not public, this change is a breaking change.

**Comment by philprime** [comment]
I believe this should be fixed with #3339. WDYT?

**Comment by philippsuess** [comment]
Yes, https://github.com/getsentry/sentry-cocoa/issues/3339 would fix this, but it's highly likely that some people rely on transactions going through beforeSend. So strictly speaking, this issue is a breaking change.

**Comment by philprime** [comment]
Yes, that's why both issues are marked as `breaking-change` and planned for the next major release. I'll close this one in favor of #3339
